backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BitcodePhases.kt

```diff
@@ -34,6 +35,16 @@ internal val contextLLVMSetupPhase = makeKonanModuleOpPhase(
             val llvmModule = LLVMModuleCreateWithNameInContext("out", llvmContext)!!
             context.llvmModule = llvmModule
             context.debugInfo.builder = LLVMCreateDIBuilder(llvmModule)
+            context.debugInfo.compilationUnit = if (context.shouldContainLocationDebugInfo()) DICreateCompilationUnit(
+                    builder = context.debugInfo.builder,
+                    lang = DWARF.language(context.config),
+                    File = "kotlin-file-${Random.nextLong()}",
```

Comment:
Making compilation results non-repeatable doesn't look like a good idea. What we actually need here?

Reply:
We need uniq name for compilation unit to prevent debug info clashing. This usually happens with in compilation with libraries have interception of used files.

Reply:
Generally we have a lot of information that can be used to produce name unique enough.
For example, resulting binary path, or stable hash of compiler inputs etc.

Reply:
Compiler inputs are not unique enough though.